13-Tenant Security Solution Setup/Scripts/RemediationScripts/Remediate-DontUseNonADIdentities.ps1

```diff
@@ -0,0 +1,364 @@
+            }
+            else
+            {
+                Write-Host "Warning: Dont pass empty string array in the ObjectIds param. If you dont want to use the param, just remove while executing the command" -ForegroundColor Yellow
+                break;
+            }  
+        }
+    }
+
+    # Adding PIM api call to fetch eligible role assignment
+    try
+    {
+        # PIM api
+        $resourceAppIdUri = "https://management.core.windows.net/"
+        $rmContext = Get-AzContext
+        [Microsoft.Azure.Commands.Common.Authentication.AzureSession]
+        $authResult = [Microsoft.Azure.Commands.Common.Authentication.AzureSession]::Instance.AuthenticationFactory.Authenticate(
+        $rmContext.Account,
+        $rmContext.Environment,
+        $rmContext.Tenant,
+        [System.Security.SecureString] $null,
+        "Never",
+        $null,
+        $resourceAppIdUri); 
+
+        $header = "Bearer " + $authResult.AccessToken
+        $headers = @{"Authorization"=$header;"Content-Type"="application/json";}
+        $method = [Microsoft.PowerShell.Commands.WebRequestMethod]::Get
+
+        # API to get eligible PIM assignment
+        $armUri = "https://management.azure.com/subscriptions/$($SubscriptionId)/providers/Microsoft.Authorization/roleEligibilityScheduleInstances?api-version=2020-10-01-preview"
+        $eligiblePIMRoleAssignments = Invoke-WebRequest -Method $method -Uri $armUri -Headers $headers -UseBasicParsing
+        $res = ConvertFrom-Json $eligiblePIMRoleAssignments.Content
+
+        # Exclude MG scope assignment
+        $excludedMGScopeAssignment =  $res.value.properties | where-object { !$_.scope.contains("/providers/Microsoft.Management/managementGroups/") }
+        $pimDistinctRoleAssignmentList += $excludedMGScopeAssignment.expandedProperties.principal | Where-Object { ![string]::IsNullOrWhiteSpace($_.email) }
+        
+        # Renaming property name
+        $distinctRoleAssignmentList += $pimDistinctRoleAssignmentList | select @{N='SignInName'; E={$_.email}}, @{N='ObjectId'; E={$_.id}}, @{N='DisplayName'; E={$_.displayName}}, @{N='ObjectType'; E={$_.type }}
+    }
+    catch
+    {
+        Write-Host "Error occured while fetching eligible PIM role assignment. ErrorMessage [$($_)]" -ForegroundColor Red
+    }
+    
+    # Defining regex to filter Non AAD Identities 
+    $NonADIdentitiesPatterns = @( "(.)*#ext(.)*" )
+
+    # Defining regex to filter allowed Non AAD Identities
+    $ApprovedNonADIndentitiesPatterns = @( "(.)*_sas.msft.net#ext#@microsoft.onmicrosoft.com" )
+
+    # Creating regex to filter Non AAD Account
+    $NonADIdentitiesPattern = (('^' + (($NonADIdentitiesPatterns | foreach {[regex]::escape($_)}) -join '|') + '$')) -replace '[\\]',''
+
+    # Filtering Non AAD Identities
+    $liveAccounts = [array]($distinctRoleAssignmentList | Where-Object {$_.SignInName -and $_.SignInName.ToLower() -imatch $NonADIdentitiesPattern} )
```

Comment:
for better readability this variable can be renamed to something like liveAccountsRoleAssignments as this is actually list of roleAssignment not just accounts.

Reply:
Added